src/OpenTelemetry/Internal/Builder/ProviderBuilderServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,42 @@
+// <copyright file="ProviderBuilderServiceCollectionExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#nullable enable
+
+using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Extensions.DependencyInjection;
+
+internal static class ProviderBuilderServiceCollectionExtensions
+{
+    public static IServiceCollection AddOpenTelemetryProviderBuilderServices(this IServiceCollection services)
+    {
+        Debug.Assert(services != null, "services was null");
+
+        services.AddOptions();
```

Comment:
No warning from the missing null forgiving operator?

```suggestion
        services!.AddOptions();
```

Reply:
This class is internal and `services` is not decorated as nullable (NOT `IServiceCollection? services`) so really, it shouldn't warn. You would get a warning at the call site if you try to pass a `null` into it. For `net6.0` & `netstandard2.1` targets, it works fine. For `net462` & `netstandard2.0` it doesn't warn on this line but it does warn on the two de-references below this! That is all kinds of wrong 🤣 Nullable analysis just doesn't work well on anything before `netstandard2.1`. So a lot of these `!`s are there just to make the IDE happy for old targets. In contrib I did [this](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/cbed25e801fb6aa7a60ecc73b16f4c9cfc044f92/build/Common.props#L41-L44). That turns off nullable warnings but then you have to have some newer target available to vet valid warnings. You get weirdness like [this](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/cbed25e801fb6aa7a60ecc73b16f4c9cfc044f92/src/OpenTelemetry.Extensions/OpenTelemetry.Extensions.csproj#L5) where it is adding targets just for analysis.